cxx4opencl/address_spaces.txt

```diff
@@ -332,9 +332,17 @@ __global const int& f(__global float &ref) {
 
 Construction, initialization and destruction of objects in `+__private+`
 and `+__global+` address space follow the general principles of {cpp}. For
-program scope objects, the implementation (i.e. compiler) defines an ABI
-format for initialization and destruction of global objects before/after
-all kernels are enqueued.
+program scope objects or static objects in the function scope with non-trivial
+constructors and destructors, the implementation defines an ABI format for
+runtime initialization and destruction of global objects before/after all
+kernels are enqueued.
+
+Non-trivial destructors with global objects is not required to be supported
+by all implementations. The predefined macro `__opencl_cpp_global_destructor`
+can be used to check whether destruction of such objects is supported
+automatically. If destructors are supported this macro will be defined and if
+they are not supported the macro is not defined during the compilation of
+kernel code.
```

Comment:
This paragraph could be slightly simplified by using "if and only if".
```suggestion
by all implementations. The predefined macro `__opencl_cpp_global_destructor`,
which is defined if and only if such destructors are supported by the implementation,
can be used to check whether this functionality is available when compiling kernel code.
```
What do you think?